### ISSUE TITLE
[Hold for Payment 2024-09-03][$250] [Pending/Scanning] The count of expenses on a report is not calculating correctly for card transactions

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#42831]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.77-4
**Reproducible in staging?:**  need reproduction
**Reproducible in production?:** need reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @kevinksullivan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1717026569202759

Details on simulating card transactions [below](https://github.com/Expensify/App/issues/42831#issuecomment-2207497832)

## Action Performed:

1. Set up workspace and enable workflows
2. Navigate to workflows and enable approvals
3. Assign expensify card to a member on that workspace
4. Incur a card transaction
5. Incur another card transaction

 
## Expected Result:
The expense count should show as `total# expenses (x scanning, y pending)`

## Actual Result:
The expense count is 0

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/1fc5eba3-4607-4125-b24c-ea4da4a1ef8b">

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0179ad48857375d973</li>
        <li>Upwork Job ID: 1810354497668334430</li>
        <li>Last Price Increase: 2024-08-19</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @getusha</details>

**Comment by mallenexpensify** [comment]
Internal since you need a card to test. @trjExpensify , I added to  #wave-collect  and set status as Release 2 (since the roadmap doc said 'card support' was then)

**Comment by trjExpensify** [comment]
Yep! CC: @kevinksullivan @grgia

**Comment by mallenexpensify** [comment]
Waiting to get picked up internally

**Comment by mallenexpensify** [comment]
@trjExpensify what's the (project) priority for this?  Considering how hard I want to push to find an internal volunteer. Thx

**Comment by trjExpensify** [comment]
Pending/Scanning is high for the summer release. This pending/scanning project is still in dev, so I'm not sure if @grgia's recent work will catch this or it needs to be looked into separately. If it's a frontend issue, I wonder if there's instructions for a contributor to simulate pending card transactions on dev to test we could provide.

**Comment by grgia** [comment]
Is this still reproducible? This most likely is a FE issue

**Comment by grgia** [comment]
Assigning to keep track of this for now

**Comment by mallenexpensify** [comment]
[Posted in #expensify-open-source
](https://expensify.slack.com/archives/C01GTK53T8Q/p1719968749799399)
> Does anyone know if we have instructions for a contributor to simulate pending card transactions on dev?  I couldn't find anything in SO or searching this room (it's for [this GH](https://github.com/Expensify/App/issues/42831#top)) (edited)

**Comment by mallenexpensify** [comment]
Since it looks like this might be FE, I'm going to open it up. 
Deets on simulating card transactions is [here in the #expensify-open-source Slack room](https://expensify.slack.com/archives/C01GTK53T8Q/p1719974823748269?thread_ts=1719968749.799399&cid=C01GTK53T8Q)
<img width="990" alt="image" src="https://github.com/Expensify/App/assets/22508304/37ce838b-e9a7-4b3a-ac15-f4b9f7d223ab">

**Comment by grgia** [comment]
@mallenexpensify could you send an update of where we're at with this issue?

**Comment by mallenexpensify** [comment]
Convo in #qa got derailed a bit cuz of 
> In short words. Our test account that is linked to our Expensify card is taking a LOT to load in New Expensify.

[Checking again about the retest. ](https://expensify.slack.com/archives/C9YU7BX5M/p1720732504607769?thread_ts=1720457243.115109&cid=C9YU7BX5M)

**Comment by isagoico** [comment]
I'm not able to reproduce this issue on my side. 
![image](https://github.com/user-attachments/assets/cf262e79-feaf-4e67-9fb2-423a52f0101f)
![image](https://github.com/user-attachments/assets/981bbca8-1b9b-4da9-b773-fe0acdf55f5c)

The correct amount of pending expenses was displayed in the preview.

**Comment by mallenexpensify** [comment]
@kevinksullivan can you test again to see if you're able to repro?  I'm unsure how to test cards

**Comment by grgia** [comment]
waiting on retest

**Comment by kevinksullivan** [comment]
here is another example:


<img width="435" alt="image" src="https://github.com/user-attachments/assets/2e137107-4c36-4f21-985b-7b5b4f4bbb11">


There are only 6 expenses on the report, and only one is shown as pending. 

This is on kevin@expensify.com



Here is the report in OldDot

<img width="824" alt="image" src="https://github.com/user-attachments/assets/78c4e9ad-3908-4209-a952-11ff81dafcd4">


ReportID: R00U97HvvgOn

**Comment by grgia** [comment]
Hmm weird, let's see how we're calculating these

**Comment by mallenexpensify** [comment]
> Hmm weird, let's see how we're calculating these

@grgia , any update on the above?

**Comment by grgia** [comment]
I haven't had a chance to look at this one yet, I'm OOO thurs/fri but will be able to make any required internal changes on Monday.

@getusha are you able to confirm whether this is a FE or BE bug?

**Comment by getusha** [comment]
Will look into this first thing tomorrow!

**Comment by trjExpensify** [comment]
@getusha did you confirm if this is a frontend or backend bug?

**Comment by getusha** [comment]
>@getusha did you confirm if this is a frontend or backend bug?

Following the dev steps i wasn't able to reproduce it, it's showing me the correct amount of pending cards, It might be a BE issue if it still persists with a real card.

https://github.com/user-attachments/assets/9b000f93-75e0-4bde-9172-0fa5f960b3f6

cc @grgia

**Comment by mallenexpensify** [comment]
Posted in the Slack thread to check to see who else is using the card for real world transactions, ie. free lunch

**Comment by grgia** [comment]
@mallenexpensify yeah if we can test this with a real card that would be great.

**Comment by mallenexpensify** [comment]
>  if we can test this with a real card that would be great.

@grgia any idea how to do this? I'm more than fine with buying some candy for the bank on my Expensify card, I'm unsure if that'd help here cuz of the first two steps

1. Set up workspace with no approvals configured
2. Assign expensify card to a member on that workspace

If it needs to be a new workspace and card, how do we even go about that?

**Comment by trjExpensify** [comment]
[Tagged you](https://expensify.slack.com/archives/C036QM0SLJK/p1722376260299189?thread_ts=1722277125.837349&cid=C036QM0SLJK) in the thread Kevin responded to you.

**Comment by mallenexpensify** [comment]
Thanks Tom, hadn't caught Kev's comment in the thread yet.  I'll see what i can do tomorrow. 
> FWIW, anyone can test the card transactions on @cardtest.expensify.com . That is how I would recreate this with a workspace that has no approvals.

**Comment by mallenexpensify** [comment]
Checking on [in thread](https://expensify.slack.com/archives/C036QM0SLJK/p1722459344040909?thread_ts=1722277125.837349&cid=C036QM0SLJK)

> for @cardtest.expensify.com , do you have more details or a link for how that works?
 I see the [How do I test the Expensify Card on staging or production?](https://stackoverflowteams.com/c/expensify/questions/3833) SO which states
> > For now, we are using fake [expensify.com](http://expensify.com/) accounts, for example you can ask a domain admin to invite [ionatan+cardtest@expensify.com](mailto:ionatan+cardtest@expensify.com) and use that account to test it.

> Is it as simple as creating a GH to get [mallen+cardtest@expensify.com](mailto:mallen+cardtest@expensify.com) invited then to request a card through New or OldDot?
>
> Found this SO too which might be helpful too
[I need to test making a real Expensify Card transaction, what do I do?](https://stackoverflowteams.com/c/expensify/questions/18010)

**Comment by mallenexpensify** [comment]
Checking [here](https://expensify.slack.com/archives/C041A90J23Z/p1722536365103979)
> I need to test an expensify card that only has a $5 limit, does anyone know where I can make multiple micro transactions so I can get multiple pending transactions?

Tried a few spots without luck. Will keep thinking.

**Comment by mallenexpensify** [comment]
Got the card setup and made 3 transactions via Paypal, using this SO, **but** I forgot to create the workspace first so I'm seeing everything in "your space"
<img width="543" alt="image" src="https://github.com/user-attachments/assets/f8fd21d1-99c5-4c10-9a8b-60691981d388">

**Comment by grgia** [comment]
I think this is because we are counting the outdated transaction objects. We need to either 1) delete the old parent transaction from ONYX entirely (reportID is set to -4) or filter these out.

**Comment by mallenexpensify** [comment]
Still struggling to reproduce this bug.  Just now followed the steps in the OP

1. Set up workspace and enable workflows
2. Navigate to workflows and enable approvals
3. Assign expensify card to a member on that workspace
4. Incur a card transaction
5. Incur another card transaction

This is what  I see
<img width="958" alt="image" src="https://github.com/user-attachments/assets/6d32aa64-f4e5-41a1-94bb-a9d8ca755d14">

**Comment by grgia** [comment]
Looking into this https://expensify.slack.com/archives/C036QM0SLJK/p1723032031245939?thread_ts=1723021882.407309&cid=C036QM0SLJK

**Comment by mallenexpensify** [comment]
I've tried testing using Paypal twice and haven't been able to reproduce. I thought I might be able to test today, by purchasing three items at a coffee shop on my Expensify Card but it's not showing in NewDot, only OldDot :/ 

<img width="1022" alt="image" src="https://github.com/user-attachments/assets/3b6d1562-633f-4b30-97fd-e29c047851b2">

I 'hacked' my way into view it in NewDot via the report ID, 7857663122685204,  and in Safari and the transactions are showing as 3 separate. 

<img width="660" alt="image" src="https://github.com/user-attachments/assets/da5b2da7-c5f3-4495-a700-8685a8a42c72">

**Comment by trjExpensify** [comment]
>I thought I might be able to test today, by purchasing three items at a coffee shop on my Expensify Card but it's not showing in NewDot, only OldDot :/

That workspace isn't supported officially on NewDot yet, because it's our Inc policy.

**Comment by grgia** [comment]
I believe this will be fixed by https://github.com/Expensify/Auth/pull/11941

**Comment by grgia** [comment]
Now that the auth PR has been deployed, let's retest this.

**Comment by grgia** [comment]
Actually let's wait on the web PR first https://github.com/Expensify/Web-Expensify/pull/43079

**Comment by grgia** [comment]
I created 2 transactions on Expense Report #707961321601087 on georgia@cardtest.expensify.com for testing. 

$2.00 - TransactionID = 5681937797917259291
$1.00 - TransactionID = 9167341062096951689

**Comment by grgia** [comment]
https://github.com/user-attachments/assets/ad34419b-b71a-453a-8441-184a6f7edfd2


Looks like this is just a simple FE bug where we shouldn't show the "0 expense" part if it's 0

**Comment by grgia** [comment]
https://github.com/Expensify/App/blob/main/src/languages/en.ts#L754-L755 We can just add a 0 check to this line

**Comment by grgia** [comment]
Confirming here https://expensify.slack.com/archives/C049HHMV9SM/p1723809488866359?thread_ts=1717026569.202759&cid=C049HHMV9SM

**Comment by grgia** [comment]
I think we can make it a bit clearer by adding a semi-colon. i.e. Confirming with marketing in the above slack convo 

`4 expenses: 1 pending, 1 scanning` - incl. 2 posted expenses
`2 expenses: 1 pending, 1 scanning`  - no posted expenses
`2 expenses - just 2 expenses`

**Comment by mike14c** [comment]
# Proposal

## Please re-state the problem that we are trying to solve in this issue.
Expense total doesn't include pending or scanning expenses.

## What is the root cause of that problem?

You are subtracting the pending and scanning expenses from the total here:

https://github.com/Expensify/App/blob/9940e99abca5b0e7f17c8d29a66df50b2345b8a2/src/components/ReportActionItem/ReportPreview.tsx#L368

## What changes do you think we should make in order to solve the problem?

```
             supportText: translate('iou.expenseCount', {
                count: numberOfRequests,
                 scanningReceipts: numberOfScanningReceipts,
                 pendingReceipts: numberOfPendingRequests,
             }),
```
if you also decide to add a semi colon then update the copy with a semi colon: 

https://github.com/Expensify/App/blob/main/src/languages/en.ts#L754-L755

**Comment by mike14c** [comment]
Contributor details
Your Expensify account email: m_capesy@live.co.uk
Upwork Profile Link: https://www.upwork.com/freelancers/~01e0f61d21e1aedd17

**Comment by grgia** [comment]
We've landed on `total# expenses (x scanning, y pending)` as the final pattern

**Comment by mallenexpensify** [comment]
Thanks @grgia , I updated the Expected behaviour in the OP to state
> The expense count should show as `total# expenses (x scanning, y pending)`

**Comment by grgia** [comment]
Opened a PR [here](https://github.com/Expensify/App/pull/47745)

**Comment by mallenexpensify** [comment]
Expected Result:
The expense count should show as total# expenses (x scanning, y pending)

What I see when testing via this SO with a paypal account on an expensify card. 
<img width="425" alt="image" src="https://github.com/user-attachments/assets/ff89babd-5a64-4dd0-a241-21533f254973">

So I think we're good here... right!??!

**Comment by mallenexpensify** [comment]
PR hit production a few days ago [link](https://github.com/Expensify/App/pull/47745#issuecomment-2311253425)

@getusha can you confirm it's fixed ?  Thx

**Comment by getusha** [comment]
>So I think we're good here... right!??!

@mallenexpensify yes i think so, all the transactions are pending right (no scanning expense)? before the fix it was showing `0 expenses (3 pending)`

**Comment by mallenexpensify** [comment]
@getusha can you please accept the job and reply here once you have?  
https://www.upwork.com/jobs/~01b3d83125b1d4f841

We need to pay via Upwork cuz you were assigned before your eligibility date for NewDot payments.

**Comment by getusha** [comment]
@mallenexpensify this was created after 2024-05-25 (my eligibility date) so we can use this to knock the remaining $250 i owe https://github.com/Expensify/App/issues/32699#issuecomment-2285081703

**Comment by mallenexpensify** [comment]
@getusha , for this issue, I'm seeing 
- assigned [getusha](https://github.com/getusha) [on Jul 3](https://github.com/Expensify/App/issues/42831#event-13387388223)  - here

And, yes, I see your eligibility date of 2024-05-25

**Comment by mallenexpensify** [comment]
@getusha can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~01b3d83125b1d4f841

Also, can you please confirm you have not been paid for this issue via NewDot and that you cancelled any requests you have for payment via NewDot?  Thx

**Comment by getusha** [comment]
@mallenexpensify accepted

>Also, can you please confirm you have not been paid for this issue via NewDot and that you cancelled any requests you have for payment via NewDot? Thx

Yes i haven't been paid via ND

**Comment by mallenexpensify** [comment]
Contributor+: @getusha paid $250 via Upwork  

Lastly.. can you fill out the checklist below? I imagine we want a regression test for this, right? 

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@] Determine if we should create a regression test for this bug.
- [ ] [@] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@BZ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:
- [ ]

**Comment by getusha** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/34191
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **Looks like it was by design at that moment.**
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/a**
- Determine if we should create a regression test for this bug. **I don't think there is an easy way to test this for QA**

**Comment by mallenexpensify** [comment]
@getusha can you propose regression test steps plz? I agree with you that QA might not be able to test but, if not, I wan to know why so I can possibly update their process/access to allow for testing. Thx

**Comment by getusha** [comment]
@mallenexpensify QA wasn't able to make transaction using expensify card check this https://github.com/Expensify/App/pull/47745#issuecomment-2306948868

**Comment by mallenexpensify** [comment]
Thanks @getusha I'm going to leave this open for a bit and mark me as the owner.  Working on an internal issue for getting C+ (and possibly QA) access to cards for testing
- https://github.com/Expensify/Expensify/issues/423994#issue-2492567759

**Comment by grgia** [comment]
@mallenexpensify can we close this one

**Comment by mallenexpensify** [comment]
Yup!!!! 📕
